tests/functional/aws-node-sdk/test/versioning/objectGet.js

```diff
@@ -0,0 +1,136 @@
+                return s3.deleteBucket({ Bucket: bucket }, done);
+            });
+        });
+
+        describe('get a delete marker', () => {
+            beforeEach(function beforeEachF(done) {
+                s3.deleteObject({ Bucket: bucket, Key: key },
+                  (err, data) => {
+                      _assertNoError(err, 'deleteObject');
+                      this.currentTest.deleteVersionId = data.VersionId;
+                      done(err);
+                  });
+            });
+
+            it('should not be able to get a delete marker', function itF(done) {
+                s3.getObject({
+                    Bucket: bucket,
+                    Key: key,
+                    VersionId: this.test.deleteVersionId },
+                err => {
+                    _assertError(err, 405, 'MethodNotAllowed');
+                    done();
+                });
+            });
+
+            it('should not be able to get a delete marker without ' +
+            'mentioning versionID ', done => {
+                s3.getObject({
+                    Bucket: bucket,
+                    Key: key },
+                err => {
+                    _assertError(err, 404, 'NoSuchKey');
+                    done();
+                });
+            });
+        });
+
+        describe('getting a object that has been deleted', () => {
+            beforeEach(function beforeEachF(done) {
+                s3.putObject({ Bucket: bucket, Key: key }, (err, data) => {
+                    _assertNoError(err, 'putObject');
+                    this.currentTest.versionId = data.VersionId;
+                    s3.deleteObject({ Bucket: bucket, Key: key },
+                      (err, data) => {
+                          _assertNoError(err, 'deleteObject');
+                          this.currentTest.deleteVersionId = data.VersionId;
+                          done(err);
+                      });
+                });
+            });
+
+            it('should not be able to get a delete marker', function itF(done) {
+                s3.getObject({
+                    Bucket: bucket,
+                    Key: key,
+                    VersionId: this.test.deleteVersionId },
+                err => {
+                    _assertError(err, 405, 'MethodNotAllowed');
+                    done();
+                });
+            });
+
+            it('should be able to get an object using its versionId',
+            function itF(done) {
+                s3.getObject({
+                    Bucket: bucket,
+                    Key: key,
+                    VersionId: this.test.versionId },
+                (err, data) => {
+                    _assertNoError(err, 'getObject');
+                    assert.strictEqual(data.VersionId, this.test.versionId);
+                    done();
+                });
+            });
+
+            it('should not be able to get an object that has been delete ' +
```

Comment:
it should return NoSuchKey if get object without version and latest version is a delete marker